sim-cli/src/main.rs

```diff
@@ -199,3 +217,47 @@ async fn main() -> anyhow::Result<()> {
 
     Ok(())
 }
+
+async fn read_sim_path(data_dir: PathBuf, sim_file: String) -> anyhow::Result<PathBuf> {
+    let sim_path = data_dir.join(sim_file);
+
+    let sim_path = if sim_path.extension().is_none() {
+        sim_path.with_extension("json")
+    } else {
+        sim_path
+    };
+
+    if sim_path.exists() {
+        Ok(sim_path)
+    } else {
+        let sim_files: Vec<String> = std::fs::read_dir(data_dir.clone())?
+            .filter_map(|f| {
+                f.ok().and_then(|f| {
+                    if f.path().extension()?.to_str()? == "json" {
+                        return f.file_name().into_string().ok();
+                    }
+                    None
+                })
+            })
+            .collect::<Vec<_>>();
+
+        if sim_files.is_empty() {
+            anyhow::bail!("no simulation files found in {:?}.", data_dir);
+        }
+
+        let selection = dialoguer::Select::new()
+            .with_prompt("Select a simulation file")
+            .items(&sim_files)
+            .default(0)
+            .interact()?;
+
+        Ok(data_dir.join(sim_files[selection].clone()))
+    }
+}
+
+async fn mkdir(dir: PathBuf) -> anyhow::Result<PathBuf> {
+    if !dir.exists() {
+        tokio::fs::create_dir(&dir).await?;
+    }
```

Comment:
This can be replaced by tokio::fs::create_dir_all

https://docs.rs/tokio/latest/tokio/fs/fn.create_dir_all.html

Reply:
simplified